### ISSUE TITLE
BitFit: Simple Parameter-efficient Fine-tuning for Transformer-based Masked Language-models

### ISSUE THREAD

**Issue by noowad93** [noowad93/ml-paper-summaries#23]
## 개요

- Transformer 기반 pretrained LM의 효율적인 파인튜닝 방법을 모색
- 본 논문에서는 task-specific한 파인튜닝을 할 때, Transformer의 일부 모듈의 bias terms과 task-specific linear layer만 파인튜닝하고 나머지 파라미터는 고정시켜도 거의 성능 하락이 없다는 것을 발견.

## Background

- 본 논문에서 fine-tuning-per-task에서 만족하고자 하는 property는 다음과 같음
    1. fully fine-tuning하는 것과 거의 동일한 수준의 성능
    2. 모델 파라미터 중 일부만을 변경할 것
    3. 모든 데이터셋에 동시에 접근할 필요 없이 태스크가 스트림에 도착할 것
    4. 변경할 파라미터의 셋이 태스크에 상관없이 동일할 것.
- pre-trained LMs를 파인튜닝하는 과정을 어떻게 해석할 것인가?
    1. learning of new capabilities
    2. exposing of existing capabilities
    - 만약 파인튜닝 과정을 exposure of existing capabilities로 해석할 수 있다면, 전체 파라미터 공간을 파인튜닝하는 방식이 아닌 효율적인 파인튜닝이 가능하지 않을까?
- Existing approaches
    - Adapter(Houlsby et al., 2019)
        - pretrained weight는 고정해두고, Adapter라는 새로운 모듈을 추가해 Adapter의 파라미터만 파인튜닝하는 방식

     <img width="629" alt="_2021-05-24__2 46 04" src="https://user-images.githubusercontent.com/49383323/119310458-ba3db300-bcaa-11eb-9d3d-fc3b6b83eb85.png">

    - Diff-Pruning (Guo et al., 2020)
        - pretrained weight는 고정해두고, sparse/task-specific한 difference-vector를 추가하여 파인튜닝하는 방식

## Bias-terms Fine-tuning (BitFit)

- 전체 파라미터의 0.1%미만을 차지하는 bias terms과 task-specific linear layer만 추가학습하여 파인튜닝하는 방식
- 학습되는 bias term
    1. l번째 레이어의 m번째 self-attention head (m,l)에 대해 아래의 q,k,v 인코더에 속하는 bias terms
        
    <img width="268" alt="2" src="https://user-images.githubusercontent.com/49383323/119310482-c32e8480-bcaa-11eb-947e-0e2bf9389fde.png">

    2. i번째 레이어의 MLP에 속하는 bias terms

     <img width="292" alt="3" src="https://user-images.githubusercontent.com/49383323/119310493-c9246580-bcaa-11eb-8753-2c32d9a73cff.png">

    3. Layer Norm의 affine transformation에 속하는 bias terms

    <img width="273" alt="4" src="https://user-images.githubusercontent.com/49383323/119310515-cfb2dd00-bcaa-11eb-9b57-76ad193c0a32.png">

- 특히, 실험적으로 query vector에 대한 bias term과 second MLP layer의 bias term만 학습시켜도 competitive한 성능을 낼 수 있었다고 함.

## Experiments

- GLUE 데이터셋에 대해 검증
- standard fine-tuning/Diff-Prune 방법론을 베이스라인으로 이용
- pre-trained BERT-base,Bert-large, RoBERTa 사용

### 결과

<img width="679" alt="5" src="https://user-images.githubusercontent.com/49383323/119310550-d93c4500-bcaa-11eb-8cf0-6aba611df19c.png">

- 더 적은양의 파라미터만 파인튜닝하고도 Diff-Pruning보다 더 좋은 성능을 보이고, Full fine-tuning과 비슷한 수준의 성능을 보이는 것을 확인.

<img width="681" alt="6" src="https://user-images.githubusercontent.com/49383323/119310557-dc373580-bcaa-11eb-928c-e1dca4362c7c.png">

- backbone 모델에 상관없이 BitFit이 유효한 방법론임을 확인

<img width="700" alt="7" src="https://user-images.githubusercontent.com/49383323/119310569-e0635300-bcaa-11eb-8a7b-b543744dd0eb.png">

- bias-parameter의 subset만을 파인튜닝해도 유효한지 확인해봄.
    - BitFit을 통해 파인튜닝했을 때의 bias vector의 변화량이 큰 bias terms만을 파인튜닝해봄
        - Bias의 변화량

      <img width="185" alt="8" src="https://user-images.githubusercontent.com/49383323/119310583-e5c09d80-bcaa-11eb-8532-25001fa01fe5.png">

    - 쿼리 인코더의 bias term과 MLP의 intermediate layer (768d→3072d)의 bias term이 변화량이 제일 커서, 두 항만 파인튜닝해봤는데 그래도 충분히 괜찮은 성능으로 수렴.
- 아직까진 NN 모델의 bias terms이 예측에서 어떤 역할을 수행하는지 명확히 분석한 논문이 많지 않음→연구가 필요하다.
    - (Wang et al., 2019)에 의하면 마지막 레이어의 bias의 값이 예측에 significant한 역할을 한다고 보고된 정도.
